test/tests.py

```diff
@@ -237,50 +237,34 @@ def _():
 # (and their corresponding network namespaces).
 @test("netns-isolation")
 def _():
-    ping_bitcoind = "ip netns exec nb-bitcoind ping -c 1 -w 1"
-    ping_nanopos = "ip netns exec nb-nanopos ping -c 1 -w 1"
-    ping_nbxplorer = "ip netns exec nb-nbxplorer ping -c 1 -w 1"
-
-    # Positive ping tests (non-exhaustive)
-    machine.succeed(
-        "%s %s &&" % (ping_bitcoind, ip("bitcoind"))
-        + "%s %s &&" % (ping_bitcoind, ip("clightning"))
-        + "%s %s &&" % (ping_bitcoind, ip("lnd"))
-        + "%s %s &&" % (ping_bitcoind, ip("liquidd"))
-        + "%s %s &&" % (ping_bitcoind, ip("nbxplorer"))
-        + "%s %s &&" % (ping_nbxplorer, ip("btcpayserver"))
-        + "%s %s &&" % (ping_nanopos, ip("lightning-charge"))
-        + "%s %s &&" % (ping_nanopos, ip("nanopos"))
-        + "%s %s" % (ping_nanopos, ip("nginx"))
-    )
+    def get_ips(*services):
+        enabled = enabled_tests.intersection(services)
+        return " ".join(ip(service) for service in enabled)
 
-    # Negative ping tests (non-exhaustive)
-    machine.fail(
-        "%s %s ||" % (ping_bitcoind, ip("spark-wallet"))
-        + "%s %s ||" % (ping_bitcoind, ip("lightning-loop"))
-        + "%s %s ||" % (ping_bitcoind, ip("lightning-charge"))
-        + "%s %s ||" % (ping_bitcoind, ip("nanopos"))
-        + "%s %s ||" % (ping_bitcoind, ip("nginx"))
-        + "%s %s ||" % (ping_nanopos, ip("bitcoind"))
-        + "%s %s ||" % (ping_nanopos, ip("clightning"))
-        + "%s %s ||" % (ping_nanopos, ip("lnd"))
-        + "%s %s ||" % (ping_nanopos, ip("lightning-loop"))
-        + "%s %s ||" % (ping_nanopos, ip("liquidd"))
-        + "%s %s ||" % (ping_nanopos, ip("electrs"))
-        + "%s %s ||" % (ping_nanopos, ip("spark-wallet"))
-        + "%s %s" % (ping_nanopos, ip("btcpayserver"))
-    )
+    # These service lists are non-exhaustive
+    reachable_from_bitcoind = get_ips("clightning", "lnd", "liquidd", "nbxplorer")
+    unreachable_from_bitcoind = get_ips("lightning-loop", "btcpayserver", "spark-wallet", "nginx")
```

Comment:
`fping` makes no difference here, because we have to run a separate ping process in each tested netns. But in general, positive ping tests are indeed cheap.

I'd still vote for keeping the test unchanged:
1. `availableNetns` is symmetrised [here](https://github.com/fort-nix/nix-bitcoin/blob/c0d04b908128f40bac41b995bd00a44ee5c45016/modules/netns-isolation.nix#L32).
2. Even if a netns connection should fail, the related positive ping test that would uncover the failure will never be executed:
   We're implicitly testing all netns connections in the service tests, so these tests would fail before the ping test.
   Only the negative ping tests can meaningfully help exposing defects.

Reply:
Seems like we're talking past each other. I'm suggesting a negative ping test from lightning-loop to bitcoind, similar to how it worked before (nanopos to bitcoind, etc). But you're right that this would cost 100ms in every netns from which we'd run this.

Reply:
I would also favor symmetrizing the tests. There could always be a bug that exposes the netns from one side.